pkgs/applications/networking/instant-messengers/chatterino2/default.nix

```diff
@@ -1,49 +1,130 @@
-{ stdenv, lib, cmake, pkg-config, fetchFromGitHub, qt6, boost, openssl, libsecret }:
+{
+  stdenv,
+  lib,
+  cmake,
+  pkg-config,
+  fetchFromGitHub,
+  qt6,
+  boost,
+  openssl,
+  libsecret,
+  darwin,
+}:
+let
+  stdenv' = if stdenv.hostPlatform.isDarwin then darwin.apple_sdk_11_0.stdenv else stdenv;
+  mkPackage =
+    {
+      pname,
+      version,
+      src,
+      extraDescription ? "",
+    }:
 
-stdenv.mkDerivation rec {
-  pname = "chatterino2";
-  version = "2.5.1";
-  src = fetchFromGitHub {
-    owner = "Chatterino";
-    repo = pname;
-    rev = "v${version}";
-    hash = "sha256-c3Vhzes54xLjKV0Of7D1eFpQvIWJwcUBXvLT2p6VwBE=";
-    fetchSubmodules = true;
+    stdenv'.mkDerivation {
+      inherit pname;
+      inherit version;
+      inherit src;
+      nativeBuildInputs = [
+        cmake
+        pkg-config
+        qt6.wrapQtAppsHook
+      ];
+      buildInputs = [
+        qt6.qtbase
+        qt6.qtsvg
+        qt6.qtimageformats
+        qt6.qttools
+        qt6.qt5compat
+        boost
+        openssl
+        libsecret
+      ] ++ lib.optionals stdenv.hostPlatform.isLinux [ qt6.qtwayland ];
+
+      env.GIT_HASH = "nix-${version}";
+      cmakeFlags = [ (lib.cmakeBool "BUILD_WITH_QT6" true) ];
+      postInstall =
+        lib.optionalString stdenv.hostPlatform.isDarwin ''
+          mkdir -p "$out/Applications"
+          mv bin/chatterino.app "$out/Applications/"
+        ''
+        + ''
+          mkdir -p $out/share/icons/hicolor/256x256/apps
+          cp $src/resources/icon.png $out/share/icons/hicolor/256x256/apps/chatterino.png
+        '';
+      passthru.updateScript = ./update.sh;
+      meta = {
+        description = "Chat client for Twitch chat";
+        mainProgram = "chatterino";
+        longDescription =
+          ''
+            Chatterino is a chat client for Twitch chat.
+            It aims to be an improved/extended version of the Twitch web chat.
+            Chatterino 2 is the second installment of the Twitch chat client series "Chatterino".
+          ''
+          + extraDescription;
+        homepage = "https://github.com/${src.owner}/${src.repo}";
+        changelog = "https://github.com/${src.owner}/${src.repo}/blob/${src.rev}/CHANGELOG.md";
```

Comment:
Those should be clickable by reviewers and be probably moved a level up

Reply:
I was not aware, but I agree it would probably make sense for the homepage to be clickable, however what influenced me to do this was that changelog in most cases contain the version in some way or another. 

This implementation was a less verbose way to provide accurate changelog/homepage links. 
I could move this up a level as you suggested to at least make the homepage clickable, however the changelog would still be having `${src.rev}` to remain accurate. This would also introduce 6 additional lines of code for these links. 
What are your thoughts on this, should i change this?